or_benchmarks/src/CheckerResultModule.cpp

```diff
@@ -0,0 +1,243 @@
+#include <benchmarks/CheckerResultModule.h>
+#include <iostream>
+#include <chrono>
+#include <ctime>
+#include <string>
+#include <exception>
+#include <algorithm>
+
+using namespace benchmarks;
+
+/*
+OpenRAVE Plugin that takes the log of a collision checker
+in the format specified by or_stub_checker and replays
+it with the collision checker of the environment.
+The time and result is sent to the output stream
+*/
+
+
+
+CheckerResultModule::CheckerResultModule(OpenRAVE::EnvironmentBasePtr env)
+    : ModuleBase(env) {
+
+    RegisterCommand("EvaluateCheck", boost::bind(&CheckerResultModule::EvaluateCheck, this, _1, _2),
+                    "Time the collision check");
+}
+
+CheckerResultModule::~CheckerResultModule() {
+
+}
+
+bool CheckerResultModule::EvaluateCheck(std::ostream &sout, std::istream &sin) {
+
+    // Obtain checker for environment
+    OpenRAVE::CollisionCheckerBasePtr checker_ptr = GetEnv() -> GetCollisionChecker();
+
+    if (!checker_ptr)
+    {
+        throw std::runtime_error("Environment does not have valid collision checker!");
+    }
+
+    // Result to output
+    std::chrono::time_point<std::chrono::high_resolution_clock> start,end;
+    double time_result = 0.0;
+    bool check_result = false;
+
+
+    // Now read the JSON string and parse it
+    std::string check_log_json_string;
+    std::getline(sin,check_log_json_string);
```

Comment:
This will break if the JSON contains newlines. I suggest [using an `istreambuf_iterator`](http://stackoverflow.com/a/3203502/111426), instead.